monkey/monkey_island/cc/ui/src/components/utils/MasqueradeUtils.js

```diff
@@ -0,0 +1,36 @@
+const MASQUERADE_STRINGS_PREFIX = 'InfectionMonkeyMasquePrefix';
+
+export function transformStringsToBytes(stringsArray){
+  const encoder = new TextEncoder('utf-8');
+
+  let bytes = stringsArray
+    .map(str => str ? encoder.encode(str + '\0') : [])
+    .reduce((acc, curr) => [...acc, ...curr], []);
+
+  let prefixBytes = encoder.encode(MASQUERADE_STRINGS_PREFIX);
+  return new Uint8Array([...prefixBytes, ...bytes]);
+}
+
+export function getStringsFromBytes(bytesArray) {
+  const encoder = new TextEncoder('utf-8');
+  const decoder = new TextDecoder('utf-8');
+  const prefixBytes = encoder.encode(MASQUERADE_STRINGS_PREFIX);
+  const uint8Array = new Uint8Array(bytesArray);
+  const prefixIndex = uint8Array.findIndex((_value, index) => {
+  for (let i = 0; i < prefixBytes.length; i++) {
+      if (uint8Array[index + i] !== prefixBytes[i]) {
+        return false;
+      }
+    }
+    return true;
+  });
+
+  if (prefixIndex === -1) {
+    return [];
+  }
+  const dataViewArray = new DataView(bytesArray, prefixIndex + prefixBytes.length);
+  const lastString = decoder.decode(dataViewArray);
```

Comment:
How do you get the last string from this?

Reply:
What is `dataViewArray`? 

> What I am getting there is mask_1\x00mask_2\x00

That makes the last string "mask_2", doesn't it?

Reply:
nope, the lastString refers to the masque Strings from the masque bytes. So first I try to find the index of the masque prefix then get everything after it (lastString, such a bad naming). decode that, split it by null and so on.

Reply:
Got it. The names are very misleading. Let's fix them.